src/components/study-manager.js

```diff
@@ -551,16 +557,71 @@ const StudyManager = ({ onClick }) => {
         };
     }, [connectNotificationsUpdateStudies, dispatchStudies]);
 
+    function addCreationRequest({ studyName, userId, ...rest }) {
+        setlocalCreationRequests({
+            ...localCreationRequests,
+            ...{
+                [makeKey({ userId: userId, studyName: studyName })]: {
+                    studyName: studyName,
+                    userId: userId,
+                    ...rest,
+                },
+            },
+        });
+    }
+
+    const cleanlocalCreationRequests = useCallback(
+        (remote) => {
+            if (localCreationRequests) {
+                let didDelete = false;
+                remote.forEach((study) => {
+                    if (localCreationRequests.hasOwnProperty(makeKey(study))) {
+                        didDelete = true;
+                        delete localCreationRequests[makeKey(study)];
+                    }
+                });
+                if (didDelete) setlocalCreationRequests(localCreationRequests);
```

Comment:
Does this do anything ? React uses shallow equality on arrays to check so it should be a noop? Maybe that's why we see the duplicate study for a split second ?

Reply:
changed